clients/application-amqp/src/main/java/org/eclipse/hono/application/client/amqp/ProtonBasedApplicationClient.java

```diff
@@ -62,6 +68,18 @@ public ProtonBasedApplicationClient(final HonoConnection connection) {
         super(connection, SendMessageSampler.Factory.noop());
     }
 
+    @Override
+    protected void onDisconnect() {
+        // consumer close handlers shall be called upon disconnect from remote peer,
+        // so skip invocation here if disconnect is done during shutdown
+        if (!connection.isShutdown()) {
+            final Throwable error = Future.failedFuture("disconnected").cause();
```

Comment:
what is the purpose of creating a Future here?

Reply:
The aim is the create a `NoStackTraceThrowable` here. I've made that now more obvious.